src/Components/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {
+  useState, useMemo, useCallback, useRef,
+} from 'react';
+import { debounce } from 'lodash';
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person) => void;
+};
+
+export const Autocomplete: React.FC<Props> = React.memo(({
+  people, onSelected,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [focus, setFocus] = useState(false);
+
+  const searchInput = useRef<HTMLInputElement>(null);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person => person.name
+      .toLocaleLowerCase().includes(appliedQuery.toLowerCase()));
+  }, [appliedQuery]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleSelectedPerson = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    person: Person,
+  ) => {
+    event.preventDefault();
+    setQuery(person.name);
+    setFocus(false);
+
+    if (searchInput.current) {
+      searchInput.current.blur();
+    }
+
+    onSelected(person);
+  };
+
+  return (
+    <div className={classNames(
+      'dropdown',
+      { 'is-active': focus === true },
+    )}
+    >
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          ref={searchInput}
+          onChange={handleQueryChange}
+          onFocus={() => setFocus(true)}
+          onBlur={() => setFocus(false)}
+        />
+      </div>
+
+      <div className="dropdown-menu" role="menu">
+        {focus
+          && (
+            <div className="dropdown-content">
+              {filteredPeople.length
+                ? filteredPeople.map(person => (
+                  <div key={person.slug} className="dropdown-item">
+                    <a
+                      href="#/"
```

Comment:
```suggestion
                      href="#"
```

Reply:
![image](https://github.com/mate-academy/react_autocomplete/assets/106619964/4f3429db-548d-4c9e-a660-c008efdce7db)

Reply:
Linter doesn`t want me to type only "#" in href.